### PR TITLE
feat(tlsn-wasm): wasm bindings

### PR DESCRIPTION
**Pull request by sinui0**
This PR implements WASM bindings for the prover and verifier. Additionally, it provides a custom test runner which handles setting up all the required dependencies to run the integration tests in the browser.

### CODE REVIEW

File: tlsn/tlsn-wasm/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    let uri = format!("https://{}/bytes?size=512", SERVER_DOMAIN);
+
+    prover
+        .setup("ws://localhost:8080/tcp?addr=localhost%3A8010")
+        .await?;
+
+    prover
+        .send_request(
+            "ws://localhost:8080/tcp?addr=localhost%3A8083",
```
Comment (heeckhau) [inline tlsn/tlsn-wasm/src/tests.rs:47]: localhost=PROXY_IP
8080=PROXY_PORT
localhost=SERVER_IP
8083=SERVER_PORT

**Review by heeckhau — CHANGES_REQUESTED** [review]
Awesome. I'm really glad you got this working and tested.

We should either hardcode the ports or use variables everywhere.

**Comment by maceip** [comment]
🙇 🦀

File: tlsn/tlsn-wasm/src/tests.rs (hunk 1 above)
Comment (heeckhau) [reply tlsn/tlsn-wasm/src/tests.rs]: 👍  let's move this to a new issue with a "good first issue" label

File: crates/wasm/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            HttpRequest {
+                method: Method::GET,
+                uri,
+                headers: HashMap::from([("Accept".to_string(), b"*".to_vec())]),
+                body: None,
+            },
+        )
+        .await?;
+
+    prover
```
Comment (heeckhau) [inline crates/wasm/src/tests.rs:58]:
```suggestion
    #[allow(clippy::single_range_in_vec_init)]
    prover
```

File: crates/wasm/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                uri,
+                headers: HashMap::from([("Accept".to_string(), b"*".to_vec())]),
+                body: None,
+            },
+        )
+        .await?;
+
+    let _ = prover.transcript()?;
+
+    prover
```
Comment (heeckhau) [inline crates/wasm/src/tests.rs:108]:
```suggestion
    #[allow(clippy::single_range_in_vec_init)]
    prover
```

**Review by heeckhau — COMMENTED** [review]
Suggested fix for Clippy warning

**Review by heeckhau — APPROVED** [review]

**Review by th4s — APPROVED** [review]
Very nice :rocket: :rocket: :rocket:

**Pull request merged**